### DISCUSSION TITLE
NextJS, useSWR hook gets undefined values on first load when accesing site with Link hook

### DISCUSSION THREAD

**Discussion by FlorjanKlancar** [vercel/next.js#31155] [Help]
TLDR: If I access page directly via the link of the page, page loads without errors; If I access page via NextJS Link hook the page throws error:

> Cannot read properties of undefined (reading '0')

I recognized that the site works normally if I go straight to the link that I want... It stops working when I'm calling components with Link tags in nextJS then it throws an error that it's in title... So if I go straight to the link everything works as expected maybe that is also the reason that the page works if I click on my link and then refresh it... So what could be the problem with Link tag?

```
Fetcher code:

const fetcher = async () => {
  const response1 = await fetch("API1");
  const data1 = await response1.json();

  const response3 = await fetch("API2");
  const data3 = await response3.json();

  const response4 = await fetch("API3");
  const data4 = await response4.json();

  const props = {
    showItemsOnOrder: data1,
    dashboardCardInfo: data3,
    dashboardGraph: data4,
  };
  return props;
};
```
Here is the useSWR part of my code:
```

export default function Dashboard(props) {
  const {data, error} = useSWR("data", fetcher);

  useEffect(() => {
    //Pusher.logToConsole = true;
    var pusher = new Pusher("pshr", {
      cluster: "eu",
    });
    const channel = pusher.subscribe("chnl");
    channel.bind("chnl", function (data) {
      console.log(data);
      mutate("data");
    });
  }, []);

  if (error) return "Error";
  if (!data) return "Loading";

  console.log(data);
  return (
    <Fragment>
      <Head>
        <title>Dashboard</title>
        <link rel="icon" href="/favicon.ico" />
       
      </Head>

      <LayoutDashboard restaurantData={props?.restaurantData[0]}>
        <DashboardPage
          orders={data?.showItemsOnOrder}
          dashboardCards={data?.dashboardCardInfo}
          ordersGraph={data?.dashboardGraph}
        />
      </LayoutDashboard>
    </Fragment>
  );
}
```
And this is the code for other component where I have Link hooks:

 ```
 <Link
       href={{
              pathname: "/restaurant/[restaurantName]/dashboard/",
              query: {restaurantName: restaurantName},
            }}
          >
            <div
              className={
                router.pathname == "/restaurant/[restaurantName]/dashboard"
                  ? "text-blue-600 bg-gray-50"
                  : "text-gray-700 "
              }
            >
              <div className="flex p-3  space-x-4 0 hover:bg-gray-50 hover:text-blue-600  cursor-pointer">
                <DonutLargeIcon className=" text-gray-300" />
                <p className="">Dashbord</p>
              </div>
            </div>
          </Link>
```

Also the **LayoutDashboard restaurantData={props?.restaurantData[0]}**
props are server side rendered so I get them everytime, the **DashboardPage** component seems to be giving me issues...
